### ISSUE TITLE
Version 2.47.0.windows.1 breaks fetch for certain repos.

### ISSUE THREAD

**Issue by SanderCokart** [git-for-windows/git#5207]
The latest version of Git hangs indefinitely on fetch on some repositories. Rolling back to the previous version fixed the issue.

- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing.

### Setup

- Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```bash
git version 2.46.2.windows.1
cpu: x86_64
built from commit: 1e20af0e809e08cca5714727dfa0fb50c0c4664d
sizeof-long: 4
sizeof-size_t: 8
shell-path: D:/git-sdk-64-build-installers/usr/bin/sh
feature: fsmonitor--daemon
libcurl: 8.10.1
OpenSSL: OpenSSL 3.2.3 3 Sep 2024
zlib: 1.3.1
```

- Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```bash
Microsoft Windows [Version 10.0.22631.4169]
```

- What options did you set as part of the installation? Or did you choose the defaults?

```bash
Editor Option: VisualStudioCode
Custom Editor Path: 
Default Branch Option: main
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Merge
Use Credential Manager: Enabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Disabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

- Any other interesting things about your environment that might be related to the issue you're seeing?

None so far.

### Details

- Which terminal/shell are you running Git from? e.g., Bash/CMD/PowerShell/other

Bash

- What commands did you run to trigger this issue? If you can provide a [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve), this will help us understand the issue.

```bash
git fetch origin
```

- What did you expect to occur after running these commands?

The fetch should complete and update the local repository.

- What actually happened instead?

The fetch command hangs indefinitely and does not complete.

We are using azure git repos and some repos we can fetch and others we can't. Rollback fixed the issue.

**Comment by dscho** [comment]
I _think_ you forgot to mention that this is fetching via SSH. In that case, this is a duplicate of #5199 and you can install the latest [snapshot](https://wingit.blob.core.windows.net/files/index.html) to fix it.

**Comment by 5bug** [comment]
I also encountered this problem. When will the official version be released to fix this issue?

**Comment by rimrul** [comment]
There is no date yet. Probably before december 16th.

**Comment by dscho** [comment]
> I also encountered this problem. When will the official version be released to fix this issue?

Please install the latest snapshot in the meantime. They are quite official.

**Comment by rescenic** [comment]
Can't fetch and pull in version 2.47. I use scoop version.

**Comment by long5436** [comment]
> # Can't fetch and pull using ssh in version 2.47. I use scoop version.
> EDIT: In my machine, I fixed it using msys-2.0.dll from GitKraken then copy & replace it to Git usr\bin folder.

Thanks for the solution, it worked for me too

**Comment by samueldcorbin** [comment]
Hope this fix will get released ASAP.

I lost a whole day trying to diagnose this specifically because I assumed that such a fundamental breakage to functionality would have meant an immediate revert or hotfix release if it were an issue with a GFW release itself, so I assumed it must be some misconfig on my end.

**Comment by dscho** [comment]
@rescenic @long5436 @samueldcorbin please, please, _please_ upgrade to the latest [snapshot](https://wingit.blob.core.windows.net/files/index.html). I _cannot_ help you if you ignore my advice.

**Comment by samueldcorbin** [comment]
@dscho I did, as soon as I finally found this issue. I'm back up and running - thanks.

But also, such a fundamental breakage should not require someone to dig up an issue on GitHub and install a snapshot, particularly when the error is so hard to diagnose/search for.

I would hope it would be reverted or hotfixed ASAP, not "eh, probably within a month". How many people are going to lose a day to this during that month?

**Comment by dscho** [comment]
@samueldcorbin see [this explanation](https://github.com/git-for-windows/git/issues/5199#issuecomment-2407462765) for the reasoning.

For all other readers stumbling over this here issue: please upgrade to the latest [snapshot](https://wingit.blob.core.windows.net/files/index.html).

**Comment by bkochwork** [comment]
This is not directly related to the SSH fetch issue but the symptoms are similar, so I hope this will be helpful to some of you.

Some subprocess of git was hanging with 100% CPU on one core. The stack trace pointed to antivirus. CrowdStrike say that this was a false positive and they have fixed it. I don't know whether the fix is global or just for our org.

Differences from #5199: Happens for push and pull and with HTTPS; doesn't happen all the time, e.g. it might work if you open a new shell window.

**Comment by themysteq** [comment]
The problem is related to OpenSSH 9.9p1 which seems to having problems with interoperability between some ssh servers. Using "external OpenSSH" (in my case - Windows 11 Enterprise) setting during installation of 2.47.0 fix the issue as it will use OpenSSH_for_Windows_9.5p1 and everything is fine. I'll add more details soon as i.e. it breaks our Gerrit 3.8.2 setup completely for Windows based users and they're unable to even work.

**Comment by shoddyguard** [comment]
Ran into another issue that is seemingly related and wanted to post here to hopefully help direct search results etc.

When using the `ssh.exe` that's bundled with git-for-windows `2.47.0` in PowerShell Core (pwsh) we noticed we were getting lockup's when attempting to redirect the output to a variable.

So for example, this works:
```
ssh myserver.mydomain.com -C ls /etc
```

However this causes pwsh to lock up:
```
$test = ssh myserver.mydomain.com -C ls /etc
```

Of note is that this only affects PowerShell Core _not_ Windows PowerShell which completes the command and correctly stores the output in the `$test` variable.

*Installing the [snapshot](https://wingit.blob.core.windows.net/files/index.html) release resolves the issue*
